vllm/core/block/block_table.py

```diff
@@ -4,6 +4,45 @@
 from vllm.utils import Device, cdiv, chunk_list
 
 
+# This class is an optimization to allow fast-access to physical block ids
+class BlockList:
```

Comment:
I think we should implement some dunder methods like `__len__`, `__getitem__`, `__setitem__` to make this class more pythonic to use (and avoid having to do some of the code changes below)

Reply:
will do

Reply:
Refactored to use __len__, __getitem__, __setitem__, looks much cleaner now.